### PR TITLE
Updates from rspec-dev (2020-12-23)

### PR DESCRIPTION
**Pull request by benoittgt**
These are some updates, generated from rspec-dev's rake tasks.

https://github.com/rspec/rspec-dev/pull/270

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,3 +42,45 @@ test case. To make this process easier, we have prepared one basic
...
+ end
++gem 'rspec-mocks', :git => "https://github.com/rspec/rspec-mocks.git", :branch => "custom-failure-message"
+```
+
+In general the process is:
+1. Create PRs explaining what you are trying to achieve.
+2. Pin the repositories to each other.
+3. Check they pass (go green).
+4. Await review if appropriate.
+5. Remove the commit from one. We will merge ignoring the failure.
```
Comment (pirj) [inline CONTRIBUTING.md:79]: I believe it is meant to be "from step 2", not one.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -42,3 +42,45 @@ test case. To make this process easier, we have prepared one basic
...
+@@ -11,6 +11,7 @@ branch = File.read(File.expand_path("../maintenance-branch", __FILE__)).chomp
+     gem lib, :git => "https://github.com/rspec/#{lib}.git", :branch => branch
+   end
+ end
++gem 'rspec-mocks', :git => "https://github.com/rspec/rspec-mocks.git", :branch => "custom-failure-message"
+```
+
+In general the process is:
+1. Create PRs explaining what you are trying to achieve.
+2. Pin the repositories to each other.
```
Comment (pirj) [inline CONTRIBUTING.md:76]: It's hard to figure out how to do this. Do you think it makes sense to mention 'maintenance-branch' here?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -42,3 +42,45 @@ test case. To make this process easier, we have prepared one basic
...
+3. Check they pass (go green).
+4. Await review if appropriate.
+5. Remove the commit from one. We will merge ignoring the failure.
+6. Remove the commit from the other, check it passes with the other commit now on main.
```
Comment (pirj) [inline CONTRIBUTING.md:80]: By the way, it won't if we delete the merged PR branch.
It's getting tricky when the change spans more than two repositories.
The best approach is to not delete PR branches until all but the last PR is merged.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -42,3 +42,45 @@ test case. To make this process easier, we have prepared one basic
...
+7. Merge the other.
+8. Rebuild the other main to check it still passes.
```
Comment (pirj) [inline CONTRIBUTING.md:82]: Can we trigger "main" by hand?
"Rebuild" sounds like "you should rebuild", while it's probably not possible for contributors. Do you think it makes sense to rephrase this to something like "we will trigger builds for the main branch of affected repositories to check if everything is in order"?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -42,3 +42,45 @@ test case. To make this process easier, we have prepared one basic
...
+
+Steps 5-8 should happen contiguously (e.g. one after another but within a short timespan)
```
Comment (pirj) [inline CONTRIBUTING.md:84]: Typo in "contiguously"

File: CONTRIBUTING.md (hunk 2 above)
Comment (pirj) [reply CONTRIBUTING.md:76]: Oh, nevermind, I've missed that we suggest pointing to the branch directly 👍

**Comment by JonRowe** [comment]
Was your source dirty? These changes don't match rspec/rspec-dev#270 ?

**Comment by benoittgt** [comment]
I think the last command run for plan text file was long time ago when I see the timestamp in file.

https://github.com/rspec/rspec-dev/blob/main/common_plaintext_files/CONTRIBUTING.md.erb#L55-L66

I pull rebase rspec-dev then `rake run["git checkout -f main && git pull origin main"]` then `rake common_plaintext_files:create_pr_with_updates["https://github.com/rspec/rspec-dev/pull/270"]`.

I will open a PR to reflect @pirj proposal on contributing doc.
